sem_covid/entrypoints/etl_dags/pwdb_worker.py

```diff
@@ -0,0 +1,190 @@
+                if CONTENT_KEY in parse_result and parse_result[CONTENT_KEY]:
+                    source[CONTENT_KEY] = parse_result[CONTENT_KEY].replace('\n', '')
+                    source[CONTENT_LANGUAGE] = (
+                            parse_result["metadata"].get("Content-Language")
+                            or
+                            parse_result["metadata"].get("content-language")
+                            or
+                            parse_result["metadata"].get("language"))
+                    valid_sources += 1
+                else:
+                    logger.warning('Apache Tika did NOT return a valid content for the source ' +
+                                   source['title'])
+
+        if valid_sources > 0:
+            logger.info('Field ' + field_data['title'] + ' had ' + str(valid_sources) + ' valid sources.')
+        else:
+            logger.warning('Field ' + field_data['title'] + ' had no valid or processable sources.')
+
+        minio.put_object_from_string(TIKA_FILE_PREFIX + hashlib.sha256(
+            (str(field_data['identifier'] +
+                 field_data['title'])).encode('utf-8')).hexdigest(), json.dumps(field_data))
+    except Exception as ex:
+        logger.exception(ex)
+
+
+def put_elasticsearch_documents(**context):
+    if "filename" not in context['dag_run'].conf:
+        logger.error(
+            "Could not find the file name in the provided configuration. This DAG is to be triggered by its parent only.")
+        return
+
+    filename = context['dag_run'].conf['filename']
+    logging.info('Processing the file ' + filename)
+
+    es_adapter = ESAdapter(config.ELASTICSEARCH_HOST_NAME,
+                           config.ELASTICSEARCH_PORT,
+                           config.ELASTICSEARCH_USERNAME,
+                           config.ELASTICSEARCH_PASSWORD)
+    logger.info('Using ElasticSearch at ' + config.ELASTICSEARCH_HOST_NAME + ':' + str(
+        config.ELASTICSEARCH_PORT))
+
+    minio = MinioAdapter(config.PWDB_COVID19_BUCKET_NAME, config.MINIO_URL, config.MINIO_ACCESS_KEY,
+                         config.MINIO_SECRET_KEY)
+    original_field_data = json.loads(minio.get_object(filename).decode('utf-8'))
+    tika_filename = TIKA_FILE_PREFIX + hashlib.sha256(
+        (str(original_field_data['identifier'] + original_field_data['title'])).encode('utf-8')).hexdigest()
+    logger.info("Tika-processed filename is " + tika_filename)
+    tika_field_data = json.loads(minio.get_object(tika_filename).decode('utf-8'))
+
+    logger.info('Sending to ElasticSearch (  ' +
+                config.PWDB_ELASTIC_SEARCH_INDEX_NAME +
+                ' ) the file ' +
+                tika_filename)
+    es_adapter.index(index_name=config.PWDB_ELASTIC_SEARCH_INDEX_NAME,
+                     document_id=tika_filename.split("/")[1],
+                     document_body=tika_field_data)
+
+    logger.info('Sent ' + tika_filename + '  to ElasticSearch.')
+
+
+default_args = {
```

Comment:
Writing DAG definitions using context blocks ("with ...")  seems like a more elegant approach. Should adopt that style?